### PR TITLE
Add TO Go server cache configs

### PR DESCRIPTION
**Pull request by rob05c**
Adds Traffic Ops Golang /servers ATS config routes.

I've parity tested against all our production servers, they all diff identical, except irrelevant ordering, and in some cases the ip_allow coalescing. 

As far as I can tell, the IP allow coalescing in Go is behaving like it should, but Perl sometimes doesn't seem to coalesce and combine like the parameters suggest. Specifically, sometimes Perl seems to combine fewer IPs than the combine parameter passed. Whoever reviews this can verify if they think it's correct.

Includes unit tests, includes changelog update. No documentation, no interface change.

- [x] This PR is not related to any other Issue

## Which Traffic Control components are affected by this PR?

- Traffic Ops

## What is the best way to verify this PR?

Generate configs, compare with Perl. This can be done with either an older TO, or personally, I change routes.go to serve Go at 1.4, causing /api/1.1 to return Perl for comparing.

## If this is a bug fix, what versions of Traffic Control are affected?

Not a bug fix.

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

**Comment by ocket8888** [comment]
> _"I've parity tested against all our production servers, they all diff identical, except irrelevant ordering, and in some cases the ip_allow coalescing"_

Are the line ordering and IP coalescence in the rewrite consistent?

**Comment by rob05c** [comment]
The line ordering isn't deterministic (because it uses a Go map), but neither is the Perl. I was planning to make all the ATS configs deterministic in the future, but I haven't done that with this or any of the other configs yet.

The IP coalescence is deterministic.

File: lib/go-atscfg/ipallowdotconfig.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+		}
+		for _, cidr := range cidr6s {
+			ipAllowData = append(ipAllowData, IPAllowData{
+				Src:    util.RangeStr(cidr),
+				Action: ActionAllow,
+				Method: MethodAll,
+			})
+		}
+
+		// allow RFC 1918 server space - TODO JvD: parameterize
```
[the remark this answers is not in the document]
Comment (rob05c) [reply lib/go-atscfg/ipallowdotconfig.go]: It is. I reproduced the comments in Perl.

File: lib/go-atscfg/servercachedotconfig.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	portStr := origin[colonI+1:]
+	port, err := strconv.Atoi(portStr)
+	if err != nil {
+		// either the port isn't an integer, or the : we found was something else.
+		// Return the origin, as if it didn't contain a port.
+		return hostName, nil
+	}
+
+	hostName = origin[:colonI]
+	return hostName, &port
```
[the remark this answers is not in the document]
Comment (rob05c) [reply lib/go-atscfg/servercachedotconfig.go]: I agree, in a future PR. As always, the intent has been to reproduce Perl as closely as possible, for safety, with the intention of improving things after it's been running stable for a bit.

**Pull request merged**